samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporter.scala

```diff
@@ -25,15 +25,18 @@ import org.apache.samza.serializers.Serializer
 import org.apache.samza.system.OutgoingMessageEnvelope
 import org.apache.samza.system.SystemProducer
 import org.apache.samza.system.SystemStream
-import org.apache.samza.util.Logging
-
+import org.apache.samza.util.{Logging, Util}
 import java.util.HashMap
 import java.util.Map
 import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit
 
 import scala.collection.JavaConverters._
 
+object MetricsSnapshotReporter {
+  val METRICS_SNAPSHOT_REPORTER_PARTITION_KEY_DEFAULT = "jobname"
```

Comment:
We use a pattern where the default value is set in config class's getXXXConfig() methods, instead of the static defaults in the implementation class.